ct/install_cmake.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```

Comment:
One thing we still might want to discuss is, what happens if the user has a cmake version beyond 3.14 installed on his system, e.g. the latest available release? 
- This could be problematic, if he requires a more recent release for some reason in different projects. This script always installs 3.14 and does not allow to keep more recent versions instead. Probably, the best solution would be to check for the cmake version using command line, and only install 3.14.7 if there is no higher version on the system.
I don't know what the best solution for this would be, however, I quickly drafted this script here, which may serve as a starting point. It may be a bit brittle, particularly the extraction of the version numbers, but it would get the job done. 
All you needed to do is to put the installation code into the right place, marked with "todo".

```
#!/bin/bash

# initialize cmake upgrade flag
execute_cmake_upgrade=0

#define minimum required cmake version
minimum_maj_version=3
minimum_min_version=14
minimum_patch_version=7

# check if cmake is installed on the system
if ! test -x /usr/bin/cmake
then
  # The cmake library is missing...
  echo "warning: This library requires cmake to be installed."
  execute_cmake_upgrade=1
else #if cmake is there
  var=$(cmake --version)      # get the currently installed cmake version
  var=${var/#cmake version /} # delete leading characters "cmake version "
  echo "Found cmake version:" $var
  vmaj=${var:0:1}   #get digits representing major version
  vmin=${var:2:2}   #get digits representing minor version
  vpatch=${var:5:2} #get digits representing patch version
  echo "extracted cmake major version" $vmaj
  echo "extracted cmake minor version" $vmin
  echo "extracted cmake patch version" $vpatch
  
  # check if the found cmake version matches the minimum requirements
  if [ $vmaj \< $minimum_maj_version ];
  then
    execute_cmake_upgrade=1
  else
    if [ $vmin \< $minimum_min_version ];
    then
      execute_cmake_upgrade=1
    else
      if [ $vpatch \< $minimum_patch_version ];
      then
        execute_cmake_upgrade=1
      fi;
    fi;
  fi;
fi;

# check if a cmake upgrade is necessary
if [ $execute_cmake_upgrade -eq 1 ]
then
  echo "cmake upgrade is required."
  #
  # TODO put function call to cmake upgrade here
  #
else
  echo "cmake version is sufficient, no upgrade required."
fi
```

Reply:
hmm yeah this is a good point. I think the script above will work, but how about just relegating the upkeep of `cmake` to a package repository which uses `apt`? We can use the official CMake apt repository: https://apt.kitware.com/

Reply:
yes, that would be an elegant, solution, too. Do you know if the provided standard version for 16.04 suits our purpose?

Reply:
Hmm, seems to be 3.17.1 if I see it correctly...

Reply:
I am not sure, but I believe it has at least 3.16.1 since it is listed under that release: https://cmake.org/download/

Reply:
Just made a commit to use the kitware repo for cmake and checked the build. It pulls in 3.17.1.